dist/rules/no-internal.js

```diff
@@ -82,21 +88,54 @@ module.exports = {
       return undefined;
     }
 
-    /** @param {string} fileName */
-    function isLocalFile(fileName) {
-      return fileName && typeof fileName === "string" && !fileName.includes("node_modules");
+    // Note: this returns false if packagePath == packageBaseDir
+    function dirContainsPath(dir, targetPath) {
+      const relative = path.relative(dir, targetPath);
+      return (
+        !!relative && !relative.startsWith("..") && !path.isAbsolute(relative)
+      );
+    }
+
+    function pathContainsCheckedPackage(packagePath) {
+      return checkedPackageRegexes.some((r) => r.test(packagePath));
     }
 
+    function owningPackageIsCheckedPackage(filePath) {
+      const packageList = workspace.getWorkspaces(filePath);
+      
+      // Look through all package infos to find the one containing our packagePath
+      let packageObj = packageList.find((pkg) => {
+        const packageBaseDir = path.dirname(pkg.packageJson.packageJsonPath);
+        return dirContainsPath(packageBaseDir, filePath);
+      });
+
+      return (packageObj !== undefined) && pathContainsCheckedPackage(packageObj.name);
+    }
+
+    // Returns true if a file is within a package for which @internal is a violation
+    // By default, @itwin and @bentley packages are included
```

Comment:
nit: could you also make this a JSDoc comment? The typescript server for javascript intellisense is much more useful if you add `@param {type} name - optional description`
for the parameters.

Some of the above function would also enjoy that being added :smile: or at least I would